enterprise/api/v1/views.py

```diff
@@ -213,6 +215,110 @@ def course_enrollments(self, request, pk):
+                    {'successes': [], 'pending': [], 'failures': []}, 409
+        """
+        enterprise_customer = self.get_object()
+        serializer = serializers.EnterpriseCustomerBulkSubscriptionEnrollmentsSerializer(
+            data=request.data,
+            context={
+                'enterprise_customer': enterprise_customer,
+                'request_user': request.user,
+            }
+        )
+        try:
+            serializer.is_valid(raise_exception=True)
+        except ValidationError:
+            return Response(serializer.errors, status=HTTP_400_BAD_REQUEST)
+
+        errors = []
+        licenses_info = serializer.validated_data.get('licenses_info')
+
+        # Default subscription discount is 100%
+        discount = serializer.validated_data.get('discount', 100.00)
+
+        # Get a list of user emails from the license info.
+        emails = {license_info['email'] for license_info in licenses_info}
+        for email in emails:
+            try:
+                validate_email_to_link(email, enterprise_customer, raise_exception=False)
+            except exceptions.ValidationError as error:
+                errors.append(error)
+
+        if errors:
+            return Response(errors, status=HTTP_400_BAD_REQUEST)
+
+        for email in emails:
+            models.EnterpriseCustomerUser.objects.link_user(enterprise_customer, email)
+
+        results = enroll_licensed_users_in_courses(enterprise_customer, licenses_info, discount)
+
+        # Retrieve all unique course key/course mode pairings from licenses_info
+        course_runs = [
+            {'key': course[0], 'mode': course[1]} for course in {
+                (info['course_run_key'], info['course_mode']) for info in licenses_info
+            }
+        ]
+        for course_run in course_runs:
+            pending_users = {
+                result.pop('user') for result in results['pending'] if result['course_run_key'] == course_run['key']
+            }
+            existing_users = {
+                result('user') for result in results['successes'] if result['course_run_key'] == course_run['key']
+            }
+            if serializer.validated_data.get('notify'):
+                enterprise_customer.notify_enrolled_learners(
+                    catalog_api_user=request.user,
+                    course_id=course_run['key'],
+                    users=pending_users + existing_users,
+                )
+
+            self._create_ecom_orders_for_enrollments(
+                course_run['key'],
+                course_run['mode'],
+                discount,
+                serializer.validated_data.get('salesforce_id'),
+                existing_users,
+            )
+        if results['failures']:
+            return Response(results, status=HTTP_409_CONFLICT)
+        if results['pending']:
+            return Response(results, status=HTTP_202_ACCEPTED)
+        return Response(results, status=HTTP_200_OK)
```

Comment:
do we need two clauses and response codes for pending vs not? if no faliures, can we always to a 202_accepted?

Reply:
I liked the idea of being able to know the state of the endpoints work just from the status code, but can go either way. @edx-abolger do you have an opinion here?

Reply:
I think the different state methods are fine but if you are going to make the status messages mean something then the 200 should be a 201 (created).